service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ManageHearingsController.java

```diff
@@ -156,9 +159,34 @@ public CallbackResponse validateHearingDatesMidEvent(@RequestBody CallbackReques
         CaseDetails caseDetails = callbackRequest.getCaseDetails();
         CaseData caseData = getCaseData(caseDetails);
 
-        List<String> errors = validateGroupService.validateGroup(caseData, HearingDatesGroup.class);
+        List<String> errors;
+
+        if (featureToggleService.isAddHearingsInPastEnabled() && (caseData.getHearingOption() == NEW_HEARING
+            || isEmpty(caseData.getHearingOption()))) {
+            errors = validateGroupService.validateGroup(caseData, PastHearingDatesGroup.class);
+        } else {
+            errors = validateGroupService.validateGroup(caseData, HearingDatesGroup.class);
+        }
+
+        if (caseData.isHearingDateInPast() && featureToggleService.isAddHearingsInPastEnabled()) {
+            caseDetails.getData().put("pageShow", YES.getValue());
```

Comment:
could this be part of `populateFieldsWhenPastDateAdded`()

Reply:
could do :) will add